tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -618,4 +625,108 @@ def test_data_loss(self):
+        rand_consumer = FranzGoVerifiableRandomConsumer(
+            self.test_context, self.redpanda, self.topic, msg_size, 10, 8,
+            self._fgo_node)
+
+        producer.start(clean=False)
+        rand_consumer.start(clean=False)
+
+        producer.wait()
+        rand_consumer.shutdown()
+        rand_consumer.wait()
+
+        # Return True if x is within the percent threshold of
+        # the given size
+        def in_percent_threshold(x, size, threshold):
+            percent_of_size = size * (threshold / 100)
+            percent_threshold = size + percent_of_size
+            self.logger.debug(f'Percent threshold: {percent_threshold}')
+            return x < percent_threshold
+
+        # Check SI cache size on all nodes
+        def check_si_cache_size():
+            is_cache_size_respected = True
+
+            for node in self.redpanda.nodes:
+                cmd = 'df --block-size=1 /var/lib/redpanda/data/cloud_storage_cache | awk \'{print $3}\' | tail -n1'
+                result = node.account.ssh_output(cmd, timeout_sec=45).decode()
+                df = float(result)
+
+                # Use -b so we get values in bytes. This flag also accounts
+                #for holes in "sparse" files since -b uses --apparent-size underneath
+                cmd = 'du -sb /var/lib/redpanda/data/cloud_storage_cache | awk \'{print $1}\''
+                result = node.account.ssh_output(cmd, timeout_sec=45).decode()
+                du = float(result)
+
+                self.logger.debug(
+                    f'{node.name}, cache-size: {self.cloud_cache_size}, df: {df}, du: {du}'
+                )
+
+                # Disk usage on si cache should be no more than 5%
+                # larger than the configured cache size. In general,
+                # 5% is an OK place to start.
+                if not in_percent_threshold(
+                        du, self.cloud_cache_size, threshold=5):
+                    is_cache_size_respected = False
+
+            return is_cache_size_respected
+
+        wait_until(check_si_cache_size,
+                   timeout_sec=30,
+                   backoff_sec=1,
+                   err_msg='Exceeded SI cache size on a node')
+
+        # Check open files count on all nodes
+        file_counts = {node.name: 0 for node in self.redpanda.nodes}
+
+        def is_open_file_count_increasing():
+            is_count_stable = True
+            for node in self.redpanda.nodes:
+                files = self.redpanda.lsof_node(node)
+                file_count = sum(1 for _ in files)
+                self.logger.debug(f'{node.name} open file count: {file_count}')
+
+                nonlocal file_counts
+                if file_count > file_counts[node.name]:
+                    file_counts[node.name] = file_count
+                    is_count_stable = False
+
+            return is_count_stable
+
+        wait_until(is_open_file_count_increasing,
```

Comment:
This is only checking that there is a 1 second period (via backoff_sec) during which open file counts don't increase, which is a kind of permissive condition.  It's also running after all the producers+consumers have stopped, so the system is idle at this point.

Checking for leaks in tests is not totally straightforward, because we expect some level of variation, but you can do something like:
- Run a consume workload that you expect to turn over the cache a known number of times (e.g. write 10GB of data each to 5 topics, then do a sequential read of each topic, so you expect the data to turn over 10 times with a 5GB cache size limit).
-  sampling every 1 second while consumes are going on, drop the first half of your samples for warmup, then require that of the remaining samples the last value is not more than a certain factor above the mean.